Lib/fontTools/merge.py

```diff
@@ -371,6 +393,69 @@ def merge(self, m, tables):
 ttLib.getTableClass('cvt ').mergeMap = lambda self, lst: first(lst)
 ttLib.getTableClass('gasp').mergeMap = lambda self, lst: first(lst) # FIXME? Appears irreconcilable
 
+@_add_method(ttLib.getTableClass('CFF '))
+def merge(self, m, tables):
+	if any(hasattr(table, "FDSelect") for table in tables):
+		raise NotImplementedError(
+			"Merging CID-keyed CFF tables is not supported yet"
+		)
+
+	newcff = tables[0]
+	newfont = newcff.cff[0]
+	private = newfont.Private
+	storedNamesStrings = []
+	glyphOrderStrings = []
+	glyphOrder = set(newfont.getGlyphOrder())
+	for name in newfont.strings.strings:
+		if name not in glyphOrder:
+			storedNamesStrings.append(name)
+		else:
+			glyphOrderStrings.append(name)
+	chrset = list(newfont.charset)
+	newcs = newfont.CharStrings
+	# XXX CharStrings object does not have a 'charStringsIndex' attribute
+	# when read from a TTX file
+	newcsi = newcs.charStringsIndex
```

Comment:
there's a `CharStrings.charStringsAreIndexed` attribute that you can test, like subset does here:
https://github.com/fonttools/fonttools/blob/b29ae8003b5ff89b5df935f70ca584ac54112fea/Lib/fontTools/subset/cff.py#L123

to support the non-indexed case, I believe you could convert the CharStrings object from indexed to non-indexed (removing the intermediation of the charStringsIndex), then treat it as non-indexed (i.e. charStrings being a Dict[str, CharString] instead of Dict[name, int])

Reply:
Is reading from TTX files an issue for this module? It does not seem to allow taking TTFont’s as input, and input file names are always passed to TTFont constructor so using TTX files as input will not work.

Reply:
I see what you mean. Well, ideally we should support both states of the CharStrings object. You can add a TODO and file an issue to track that if you don't do this in the current PR.

Reply:
Should be supported now.